test/cql-pytest/test_compactionstats.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import pytest
+
+import nodetool
+from util import new_test_table
+
+
+def get_compaction_stats(cql, table):
+    ks, cf = table.split('.')
+    stats = nodetool.compactionstats(cql)
+    tasks = 0
+    for s in stats:
+        if s['ks'] == ks and s['cf'] == cf:
+            tasks += int(s['task'])
+    return tasks, stats
+
+@pytest.mark.parametrize("compaction_strategy", ["LeveledCompactionStrategy", "SizeTieredCompactionStrategy", "TimeWindowCompactionStrategy"])
+def test_compactionstats_after_major_compaction(scylla_only, cql, test_keyspace, compaction_strategy):
+    """
+    Test that compactionstats show no pending compaction after major compaction
+    """
+    num_sstables = 16
+    extra_strategy_options = ""
+    if compaction_strategy == "LeveledCompactionStrategy":
+        extra_strategy_options = ", 'sstable_size_in_mb':1"
+        num_sstables *= 4   # Need enough data to trigger level 0 compaction
+    value = 'x' * 128*1024
+    with new_test_table(cql, test_keyspace,
+                        schema="p int PRIMARY KEY, v text",
+                        extra=f"WITH compression={{}} AND compaction={{'class':'{compaction_strategy}'{extra_strategy_options}}}") as table:
+        with nodetool.no_autocompaction_context(cql, table):
+            for i in range(num_sstables):
+                cql.execute(f"INSERT INTO {table} (p, v) VALUES ({i}, '{value}')")
+                nodetool.flush(cql, table)
+            tasks, stats = get_compaction_stats(cql, table)
+            assert tasks > 0, f"Found no pending compaction tasks as expected: stats={stats}"
```

Comment:
Hmm, isn't this test at a risk being flaky? How do you know compaction won't have happened spontaneously before this line, and right now there's no compaction tasks left because it all happened already?

Reply:
Oh, I see.